core/src/main/java/fj/Monoid.java

```diff
@@ -402,6 +402,16 @@ public A join(final Iterable<A> as, final A a) {
   }
 
   /**
+   * A monoid for the maximum of two integers.
+   */
+  public static final Monoid<Integer> intMaxMonoid = monoid(Semigroup.intMaximumSemigroup, Integer.MIN_VALUE);
```

Comment:
I'm using this monoid to annotate the nodes of the finger tree (the measured values) with integers and the max operation means the finger tree represents a priority queue.

The existing finger tree implementation uses a monoid to annotate the node, so I using it for that.